Hotfix ResourceManager to use pyvisa-py

Freddie-Burns opened the pull request:
USB powermeters are sometimes not recognised on machines where NI-VISA is installed. Forcing ResourceManager to use the pyvisa-py USB driver with '@py' fixes this issue.

Freddie-Burns committed:
Hotfix ResourceManager to use pyvisa-py

USB powermeters are sometimes not recognised on machines where NI-VISA is installed. Forcing ResourceManager to use the pyvisa-py USB driver with '@py' fixes this issue.

Freddie-Burns committed:
Alter lightrig for connecting to laser

laser_channel variable was incorrectly named.

kwargs were not unpacked with **.

'@py' should force ResourceManager to use correct USB driver allowing detection of USB devices.

laser_COM_port missing self as it is defined as an instance variable.

Simplify string formatting on line 791.

Freddie-Burns committed:
Alter Powermeter in lightrig.py to fix connection issues

Powermeter must get arguments as kwargs due to how they are unpacked and assigned to instance variables later.

foldername logic was not correct for foldernames stored as strings.

Force choice of USB driver with '@py' in ResourceManager instanciation.

Freddie-Burns committed:
Merge branch 'main' of https://github.com/Freddie-Burns/lightrig

inwwin commented:
Other commits from your main branch is polluting this PR. But let's wait for Light Trace Photonics people to comment on this.

Freddie-Burns committed:
Ignore PyCharm files

Freddie-Burns committed:
Correct lightrig import

Freddie-Burns committed:
Create csv for automated coupling

(x, y) coords for grating couplers of first two ring resonators in the line we hope to investigate.

Freddie-Burns committed:
Fill resonator positions

Add (x, y) coords for all lined up 24 ring resonators on our chip's GDS file.
